tests/components/landisgyr_heat_meter/test_sensor.py

```diff
@@ -199,3 +210,44 @@ async def test_restore_state(mock_heat_meter, hass: HomeAssistant) -> None:
     assert state
     assert state.state == "devicenr_789"
     assert state.attributes.get(ATTR_STATE_CLASS) is None
+
+
+@patch(API_HEAT_METER_SERVICE)
+async def test_exception_on_polling(mock_heat_meter, hass: HomeAssistant) -> None:
+    """Test sensor."""
+    entry_data = {
+        "device": "/dev/USB0",
+        "model": "LUGCUH50",
+        "device_number": "123456789",
+    }
+    mock_entry = MockConfigEntry(domain=DOMAIN, unique_id=DOMAIN, data=entry_data)
+    mock_entry.add_to_hass(hass)
+
+    # First setup normally
+    mock_heat_meter_response = MockHeatMeterResponse(
+        heat_usage_gj=123.0,
+        volume_usage_m3=456.0,
+        heat_previous_year_gj=111.0,
+        device_number="devicenr_789",
+        meter_date_time=dt_util.as_utc(datetime.datetime(2022, 5, 19, 19, 41, 17)),
+    )
+
+    mock_heat_meter().read.return_value = mock_heat_meter_response
+
+    await hass.config_entries.async_setup(mock_entry.entry_id)
+    await async_setup_component(hass, HA_DOMAIN, {})
+    await hass.async_block_till_done()
+    await hass.services.async_call(
+        HA_DOMAIN,
+        SERVICE_UPDATE_ENTITY,
+        {ATTR_ENTITY_ID: "sensor.heat_meter_heat_usage"},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
```

Comment:
I think it was a good idea to assert the initial setup and check one of the entities.
```suggestion
    await hass.async_block_till_done()
    
    # check if initial setup succeeded
    state = hass.states.get("sensor.heat_meter_heat_usage")
    assert state
    assert state.state == "34.16669"
```